### PR TITLE
apply inverse() in simplify

### LINKED ISSUE #10487: Apply inverse() on custom functions
See https://stackoverflow.com/questions/35060127/sympy-simplifying-of-inverse-function. If you define `inverse` on a custom function, it works with `solve`, but you can't actually reduce it, like `f(g(x))` -> `x` (if `f(x).inverse() == g`).  Probably `simplify` should do it.

### PR DESCRIPTION
**Pull request by AnishShah**
Fixes #10487

### CODE REVIEW

File: sympy/simplify/simplify.py [hunk 1]
```diff
@@ -522,6 +522,10 @@ def simplify(expr, ratio=1.7, measure=count_ops, fu=False):
         return expr
 
     if not isinstance(expr, (Add, Mul, Pow, ExpBase)):
+        if isinstance(expr, Function) and hasattr(expr, "inverse"):
+            if isinstance(expr.args[0], expr.inverse()):
```
Comment (asmeurer) [inline sympy/simplify/simplify.py]: inverse has an argindex parameter. We should be using that.

Comment (AnishShah) [inline sympy/simplify/simplify.py]: Hi @asmeurer, I searched in the docs, but couldn't find the meaning of `argindex` or how it is used in `inverse` function. Can you please tell?

Comment (AnishShah) [inline sympy/simplify/simplify.py]: @asmeurer Should I pass `argindex=0` to `inverse` as we are using `expr.args[0]` ?

Comment (asmeurer) [inline sympy/simplify/simplify.py]: It has to do with functions of multiple variables. However, I just looked in the solvers and it doesn't look like they are handling it either. I don't think inverse with argindex other than 1 is used anywhere in SymPy at this time.

**Comment by AnishShah** [comment]
@asmeurer I added `argindex=1` as in future, we might add functions of multiple variables. So, I have made some changes and added some tests.

**Comment by AnishShah** [comment]
ping @asmeurer .. any review?

**Pull request merged**